shared/rakefiles/xk_util.rake

```diff
@@ -240,15 +240,28 @@ end
 
 # This task restores a list of images in snapshots.
 task :restore_snapshot_from_image_file, [:snapshot_files] => [@gcp_creds_file, :configure_extra_tf_vars] do |taskname, args|
-  require 'csv'
+  if args[:snapshot_files].nil? || args[:snapshot_files].size == 0
+    puts "  ERROR: Argument :snapshot_files not present!"
+    raise
+  end
 
+  require 'csv'
+  # kubectl get pv -o json | jq -r '.items[] | "\(.spec.claimRef.name),\(.metadata.labels."failure-domain.beta.kubernetes.io/zone")"'
+  #
+  # database-storage-couchdb-couchdb-0,us-central1-f
+  # database-storage-couchdb-couchdb-1,us-central1-a
   pv_zones = {}
   CSV.parse(%x{
-              #{@exekube_cmd} kubectl get pv -o json | jq -r '.items[] | \"\\(.spec.claimRef.name),\\(.metadata.labels.\"failure-domain.beta.kubernetes.io/zone\")\"'
-           }.chomp).each do |line|
-             pv_zones[line[0]] = line[1]
-           end
+      #{@exekube_cmd} kubectl get pv -o json | jq -r '.items[] | \"\\(.spec.claimRef.name),\\(.metadata.labels.\"failure-domain.beta.kubernetes.io/zone\")\"'
```

Comment:
Perhaps it would be easier and more readable to directly parse the json output from `kubectl` to ruby (using smth. like `JSON.parse`) and then work with it as with object within ruby. (Instead of using `jq` and "manually" constructing CSV-like output and parsing it with another (`CSV`) library.)

Reply:
I'm very sorry of make all of you to read such a piece of shitty code. I can not understand what mind state I had when I wrote that. I hope the latest code seems like made by a real engineer.